### ISSUE TITLE
Question - How to use postcss?

### ISSUE THREAD

**Issue by mbulfair** [ng-packagr/ng-packagr#1471]
Not sure if this is the right spot, but I am looking on how to use postcss on build of a library. I use normal .css files, but I `@apply` tailwind classes (for example) to it. How can I go about extending this to support postcss?

Thank You

#### Version Information

```
$ node_modules/.bin/ng-packagr --version
ng-packagr:            5.7.1
@angular/compiler:     8.2.14
rollup:                1.25.2
tsickle:               0.37.1
typescript:            3.5.3
```

_Please include all version numbers that might be relevant, e.g. third-party libraries_

**Comment by nuriu** [comment]
I'm looking for same solution.

@mbulfair did you managed to use it?

**Comment by alan-agius4** [comment]
Currently it’s not possible. I’ll mark this as a feature request.

**Comment by gquinteros93** [comment]
Hi, I am interested in this feature too. Because I will like to use [postcss](https://postcss.org/), [postcss-css-modules](https://github.com/css-modules/postcss-modules) and [posthtml-css-modules](https://github.com/posthtml/posthtml-css-modules)  in order to use CSS Modules in my Custom Angular Library.

Thanks in advance.

**Comment by junyper** [comment]
What about using https://github.com/michael-ciniawsky/postcss-load-config to load a postcss config file (if it exists) and falling back to the defaults (the plugins that are in use now) if it's not there?

**Comment by gquinteros93** [comment]
Hi @alan-agius4, Do you know when it is going to be possible to use postcss in ng-packagr?

Could be possible to extend the rollupjs config in order to use https://github.com/egoist/rollup-plugin-postcss#readme ?

**Comment by cgatian** [comment]
From what I can gather the CSS compilation has to be synchronous. TailwindCSS is an asynchronous plugin for PostCSS, when I attempted to add this plugin to the pipeline I received the following error from PostCSS:
```bash
ERROR: Use process(css).then(cb) to work with async plugins
```
I then refactored the ngPackagr pipeline to support async and received the following error from ngcc:
```bash
ERROR: Cannot handle async resource. angularCompilerOptions.enableResourceInlining requires all resources to be statically resolvable.
```
From what I can tell there's no way to force PostCSS to process synchronously, run Tailwind in Async, or allow ngcc to resolve the resources asynchronously. Hope the maintainers can provide a little more context on this.

I think the only option is to run a separate process (using Webpack or Tailwind's CLI tool) to process all the component styles that are using Tailwind features to create a compiled stylesheet. Meaning for each component when tailwind is required create a new tailwind specific stylesheet.

Sample directory structure:
```bash
mycomp.component.ts
mycomp.component.scss  <--- component scoped styles (rarely used)
mycomp.component.html <--- can still use tailwind classes
mycomp.component.tailwind.scss <--- Uses @apply directives
```
